Userland/Libraries/LibWeb/DOM/LiveNodeList.cpp

```diff
@@ -67,11 +66,11 @@ Node* LiveNodeList::first_matching(Function<bool(Node const&)> const& filter) co
         });
     } else {
         m_root->for_each_child([&](auto& node) {
-            if (m_filter(node) && filter(node)) {
+            if (matched_node)
+                return;
+
+            if (m_filter(node) && filter(node))
                 matched_node = const_cast<Node*>(&node);
-                return IterationDecision::Break;
-            }
-            return IterationDecision::Continue;
```

Comment:
hmm, should _we_ expect an iteration decision though? 🤔 Otherwise we needlessly iterate over every child even if we've immediately found what we're looking for

Reply:
I've updated the PR so that single-level iteration methods (`for_each_child()` and `for_each_child_of_type()`) return an `IterationDecision`.